### PR TITLE
not trying to update RESV target for prediction producers

### LINKED ISSUE #1866: A bug for assuming RESV can only be history matching wells
If WCONPROD is specified with RESV control,  the following assertion failure will be triggered. 

```
OPM-devel/debug/opm-simulators/opm/autodiff/SimFIBODetails.hpp:91: void Opm::SimFIBODetails::historyRates(const Opm::PhaseUsage&, const Opm::ProductionControls&, std::vector<double>&): Assertion `! p.prediction_mode' failed.
```

It should not be difficult to fix anyway. 

Possibly other related bugs can be found related to injectors.

### PR DESCRIPTION
**Pull request by GitPaean**
this closes https://github.com/OPM/opm-simulators/issues/1866

### CODE REVIEW

File: opm/simulators/wells/BlackoilWellModel_impl.hpp [hunk 1]
```diff
@@ -1703,21 +1704,23 @@ namespace Opm {
                             }
                             const auto& wp = i->second;
                             const auto production_controls = wp.productionControls(summaryState);
```
Comment (GitPaean) [inline opm/simulators/wells/BlackoilWellModel_impl.hpp]: Not sure why `summaryState` is here. It is possible something WIP.

**Comment by GitPaean** [comment]
jenkins build this please

File: opm/simulators/wells/BlackoilWellModel_impl.hpp (hunk 1 above)
Comment (bska) [reply opm/simulators/wells/BlackoilWellModel_impl.hpp]: > Not sure why `summaryState` is here. It is possible something WIP.

It was introduced in commit 511645d12, as part of PR #1831. It's work in progress in the sense that the ACTIONX/UDQ feature is as yet incomplete, but the simulator side of the work is apparently very close to completion (see PR #1865).

**Review by bska — APPROVED** [review]
No real comments. This addresses the immediate problem. On the other hand the `computeRESV` function is becoming very complicated (many nested levels of conditionals) and might be candidate for refactoring at some point.

File: opm/simulators/wells/BlackoilWellModel_impl.hpp (hunk 1 above)
Comment (GitPaean) [reply opm/simulators/wells/BlackoilWellModel_impl.hpp]: Thanks for the explanation.

**Pull request merged**